rllib/examples/policy/cliff_walking_wall_policy.py

```diff
@@ -0,0 +1,85 @@
+from ray.rllib.policy.policy import Policy, ViewRequirement
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.typing import AlgorithmConfigDict, TensorStructType, TensorType
+from typing import Dict, Union, List, Tuple, Optional
+from ray.rllib.utils.annotations import override
+from ray.rllib.models.torch.torch_action_dist import TorchCategorical
+import numpy as np
+import gym
+
+
+class CliffWalkingPolicy(Policy):
+    """Optimal RLlib policy for the CliffWalking environment with
+    epsilon-greedy exploration"""
+
+    @override(Policy)
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        config: AlgorithmConfigDict,
+    ):
+        super().__init__(observation_space, action_space, config)
+        # Known optimal action dist for each of the 48 states and 4 actions
+        self.optimal_dist = np.zeros((48, 4), dtype=float)
+        # Starting state: go up
+        self.optimal_dist[36] = (1, 0, 0, 0)
+        # Cliff + Goal: never actually used, set to random
+        self.optimal_dist[37:] = (0.25, 0.25, 0.25, 0.25)
+        # Row 2; always go right
+        self.optimal_dist[24:36] = (0, 1, 0, 0)
+        # Row 0 and Row 1; go down or go right
+        self.optimal_dist[0:24] = (0, 0.5, 0.5, 0)
+        # Col 11; always go down, supercedes previous values
+        self.optimal_dist[[11, 23, 35]] = (0, 0, 1, 0)
+        assert np.allclose(self.optimal_dist.sum(-1), 1)
+
+        # Initialize action_dist used to compute actions as optimal_dist
+        self.update_epsilon(config.get("epsilon", 0.0))
```

Comment:
self.action_dist is not defined here. Please define it in the constructor even if it is set to None.

Reply:
I realized we didn't need the whole update_epsilon thing anymore, so I'm just initializing the action_dist in the constructor.